src/ResourceManager/KeyVault/Commands.KeyVault/Commands/RemoveAzureKeyVaultCertificate.cs

```diff
@@ -46,65 +55,83 @@ public class RemoveAzureKeyVaultCertificate : KeyVaultCmdletBase
         /// </summary>       
         [Parameter(Mandatory = true,
                    Position = 1,
+                   ParameterSetName = ByVaultNameAndNameParameterSet,
                    ValueFromPipelineByPropertyName = true,
                    HelpMessage = "Specifies the name of the certificate in key vault.")]
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
         /// <summary>
-        /// If present, do not ask for confirmation
+        /// Vault Object
         /// </summary>
-        [Parameter( Mandatory = false,
-                    HelpMessage = "Do not ask for confirmation.")]
-        public SwitchParameter Force { get; set; }
+        [Parameter(Mandatory = true,
+                   Position = 0,
+                   ParameterSetName = ByObjectParameterSet,
+                   ValueFromPipeline = true,
+                   HelpMessage = "Certificate Object.")]
+        [ValidateNotNullOrEmpty]
+        public PSKeyVaultCertificateIdentityItem InputObject { get; set; }
 
         /// <summary>
         /// If present, operate on the deleted key entity.
         /// </summary>
-        [Parameter( Mandatory = false,
-                    HelpMessage = "Permanently remove the previously deleted certificate." )]
+        [Parameter(Mandatory = false,
```

Comment:
(nitpick) I've seen that elsewhere in this changeset the InRemovedState parameter was marked as mandatory for the 'deleted' parameter set. Here it seems to be missing. Would you consider adding the 'deleted' parameter set to this cmdlet as well, and maintain consistency with other cmdlets?